### DISCUSSION TITLE
PyInstaller with generated .py files

### DISCUSSION THREAD

**Discussion by ughstudios** [pyinstaller/pyinstaller#6880] [Help]
I want to use PyInstaller to package a script into an executable. However, I am getting `importlib` errors when I try to run the file and it says it cannot find a file. The file in question is a file that is generated by another python program, is there a way to run the generator when the .exe is launched? 

Essentially, we have another program within our python code that generates a list of variables that are used throughout our application, but this generated file cannot be simply packaged, we need it to be generated when the .exe is launched.

**Comment by bwoodsend** [comment]
The rules w.r.t imports are the same for normal Python as they are for PyInstaller. The directory in which you write that dynamically generated file needs to be added to `sys.path` before importing. That said, I strongly recommend against writing code to a file then importing it in for security reasons. It sounds like you should just have a regular module which pulls in those variables then the rest of your code imports it.

```python
# dynamic_globals.py

something_dynamic = [go fetch that value from wherever it's coming from]
```
```python
# some_other_part_of_your_application.py
from dynamic_globals import something_dynamic
```
Then your back into sane territory on all fronts.

**Reply by ughstudios** [reply]
```
The rules w.r.t imports are the same for normal Python as they are for PyInstaller. The directory in which you write that dynamically generated file needs to be added to sys.path before importing. That said, I strongly recommend against writing code to a file then importing it in for security reasons. It sounds like you should just have a regular module which pulls in those variables then the rest of your code imports it.
```

I would normally agree with you but the problem is that these variables need to be generated on the fly as they can be different for different workspaces, or computer setups. 

Also, I consistently seem to get other errors if I change these, I get other exceptions like KeyError from a dictionary that does not occur when running purely through python and not using PyInstaller, and the output callstack is not very verbose. Is there a way to get a more verbose callstack?

**Reply by bwoodsend** [reply]
You'll have to give us a self contained example then. This is too vague for us to be able to tell what you're doing wrong. There are loads of ways this can go wrong, some of which are PyInstaller specific and some which PyInstaller only flushes out of otherwise unknowingly already broken code (usually working directory misassumptions or lack of understanding of Python packaging...).
